devtools/src/index.ts

```diff
@@ -25,3 +26,26 @@ export function provideQueryDevTools(
     },
   ]);
 }
+
+export function provideLazyQueryDevTools(
+  devToolOptions: Partial<Omit<TanstackQueryDevtoolsConfig, 'client'>> = {},
+  triggerFunction: string = 'toggleDevtools',
+) {
+  return makeEnvironmentProviders([
+    {
+      provide: APP_INITIALIZER,
+      multi: true,
+      useFactory() {
+        const queryClient = injectQueryClient();
+        const destroyRef = inject(DestroyRef);
+        return () => {
+          (window as any)[triggerFunction] = () => {
```

Comment:
i would advise agains marking windows as any, a good strategy here is declaration merging of the window object with your own declaration. https://www.typescriptlang.org/docs/handbook/declaration-merging.html

Reply:
True, but in this case the key is arbitrary (provided as an argument) so it would have to be merged with `[key: string]` making it not particularly useful?

Unless the ability to change the function name is removed.